packages/datetime/src/dateInput.tsx

```diff
@@ -327,6 +327,10 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
      * This is a no-op unless moment-timezone's setDefault has been called.
      */
     private fromMomentToDate = (momentDate: moment.Moment) => {
+        if (momentDate == null) {
+            return null;
```

Comment:
yeah, I would never explicitly return `null` in our code (but this is an outdated diff)